### PR TITLE
Materialize inputs en masse when executing a 0-output lazy operation.

### PR DESCRIPTION
**Pull request by bgogul**
This PR does not introduce any new functionality. 
Related to https://bugs.swift.org/browse/TF-604

### CODE REVIEW

File: Sources/TensorFlow/Core/LazyTensorOperation.swift [hunk 1]
```diff
@@ -374,9 +374,28 @@ extension LazyTensorOperation: TFTensorOperation {
...
+        var unmaterializedInputs = Array<LazyTensorOperation>()
+        unmaterializedInputs.reserveCapacity(inputs.count)
+        for input in inputs {
+            switch input {
+            case .single(let v):
+                if let lazyOperation = v.lazyTensorOperation {
+                    unmaterializedInputs.append(lazyOperation)
+                }
+            case .list(let values):
+                for value in values {
```
Comment (dan-zheng) [inline Sources/TensorFlow/Core/LazyTensorOperation.swift]: Consider using [`Array.compactMap(_:)`](https://developer.apple.com/documentation/swift/array/2957701-compactmap):
```swift
                unmaterializedInputs.append(
                    contentsOf: values.compactMap { $0.lazyTensorOperation }
                )
```

Comment (bgogul) [reply Sources/TensorFlow/Core/LazyTensorOperation.swift]: Thanks for the suggestion, @dan-zheng However, isn't the explicit loop slightly more efficient as we don't need to create the temporary array?

Comment (dan-zheng) [reply Sources/TensorFlow/Core/LazyTensorOperation.swift]: > Thanks for the suggestion, @dan-zheng. However, isn't the explicit loop slightly more efficient as we don't need to create the temporary array?

Yes, oops. You can use a [`LazySequence`](https://developer.apple.com/documentation/swift/lazysequence), which will not materialize a temporary array:
```diff
                 unmaterializedInputs.append(
-                    contentsOf: values.compactMap { $0.lazyTensorOperation }
+                    contentsOf: values.lazy.compactMap { $0.lazyTensorOperation }
                 )
```

Comment (bgogul) [reply Sources/TensorFlow/Core/LazyTensorOperation.swift]: Gotcha, thanks!

**Pull request merged**